controllers/watcher_controller.go

```diff
@@ -299,6 +300,13 @@ func (r *WatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request) (re
 			condition.ReadyCondition, condition.ReadyMessage)
 	}
 
+	topLevelSecretName, err := r.createTopLevelSecret(ctx, helper, instance, transporturlSecret, inputSecret)
```

Comment:
Shouldn't this go before setting Watcher ReadyCondition to true?, actually, i think this should be part of InputReadyCondition ?

Reply:
that makes sense, moving it to before setting `InputReadyCondition` to true